Competition2017/src/competition/subsystems/vision/VisionSubsystem.java

```diff
@@ -91,21 +90,32 @@ private void handleCommPacket(OffboardCommPacket packet) {
                 newTarget.isTracked = jsonTarget.getBoolean("isTracked");
 
                 newTarget.distance = jsonTarget.getDouble("distance");
-                
+
                 return newTarget;
             });
         }
     }
-    
-    private <T> void loadJsonArray(List<T> outList, JSONObject payload, String jsonKey, Function<JSONObject, T> objectParser) {
+
+    private <T> void loadJsonArray(List<T> outList, JSONObject payload, String jsonKey,
+            Function<JSONObject, T> objectParser) {
         outList.clear();
-        if(payload.has(jsonKey)) {
+        if (payload.has(jsonKey)) {
             JSONArray jsonTargets = payload.getJSONArray(jsonKey);
-            for(Object targetObj : jsonTargets) {
-                if(JSONObject.class.isInstance(targetObj)) {
-                    outList.add(objectParser.apply((JSONObject)targetObj));
+            for (Object targetObj : jsonTargets) {
+                if (JSONObject.class.isInstance(targetObj)) {
+                    outList.add(objectParser.apply((JSONObject) targetObj));
                 }
             }
         }
     }
+
+    @Override
+    public void updatePeriodicData() {
+        double currentTime = Timer.getFPGATimestamp();
+        if (currentTime - lastTimePacketRecieved >= 5 || packet == null) {
+            log.info("Disconnected");
+        } else {
+            log.info(packet.id + " (" + packet.packetType + "): " + packet.payload.toString());
```

Comment:
Instead of logging the most recent packet, log the following information:
- The "packets per second"
- The current number of tracked boilers (`trackedBoilers .size()`)